plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsTableHandle.java

```diff
@@ -14,50 +14,62 @@
 package io.trino.plugin.google.sheets;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.trino.spi.connector.ConnectorTableHandle;
-import io.trino.spi.connector.SchemaTableName;
 
 import java.util.Objects;
 
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 public final class SheetsTableHandle
         implements ConnectorTableHandle
 {
-    private final SchemaTableName schemaTableName;
+    private final SheetsRelationHandle relationHandle;
```

Comment:
Instead of making `SheetsTableHandle` a "two-in-one", we could add another implementation of `ConnectorTableHandle` for the purpose of the Table Function. The code would be clearer that way, and we would avoid nesting / optionality which makes it harder to reason about.

Reply:
Fixed to remove nesting, take another look.